x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -16,104 +16,242 @@ import {
+  }, [mappingsDefinition]);
+
+  useMappingsStateListener({ value: parsedDefaultValue, status: 'disabled' });
+
+  const {
+    fields: { byId, rootLevelFields },
+    search,
+    documentFields,
+  } = useMappingsState();
+
+  const getField = useCallback((fieldId: string) => byId[fieldId], [byId]);
+  const fields = useMemo(() => rootLevelFields.map(getField), [rootLevelFields, getField]);
+  const dispatch = useDispatch();
+  const onSearchChange = useCallback(
+    (value: string) => {
+      dispatch({ type: 'search:update', value });
+    },
+    [dispatch]
+  );
+
+  const searchTerm = search.term.trim();
+
+  const jsonBlock = (
+    <EuiCodeBlock
+      language="json"
+      isCopyable
+      data-test-subj="indexDetailsMappingsCodeBlock"
       css={css`
         height: 100%;
       `}
     >
-      <EuiFlexItem
-        grow={1}
-        css={css`
-          min-width: 400px;
-        `}
-      >
-        <EuiPanel grow={false} paddingSize="l">
-          <EuiFlexGroup alignItems="center" gutterSize="s">
-            <EuiFlexItem grow={false}>
-              <EuiIcon type="iInCircle" />
-            </EuiFlexItem>
-            <EuiFlexItem>
-              <EuiTitle size="xs">
-                <h2>
-                  <FormattedMessage
-                    id="xpack.idxMgmt.indexDetails.mappings.docsCardTitle"
-                    defaultMessage="About index mappings"
-                  />
-                </h2>
-              </EuiTitle>
-            </EuiFlexItem>
-          </EuiFlexGroup>
-          <EuiSpacer size="s" />
-          <EuiText>
-            <p>
-              <FormattedMessage
-                id="xpack.idxMgmt.indexDetails.mappings.docsCardDescription"
-                defaultMessage="Your documents are made up of a set of fields. Index mappings give each field a type
-              (such as keyword, number, or date) and additional subfields. These index mappings determine the functions
-              available in your relevance tuning and search experience."
-              />
-            </p>
-          </EuiText>
-          <EuiSpacer size="m" />
-          <EuiLink
-            data-test-subj="indexDetailsMappingsDocsLink"
-            href={documentationService.getMappingDocumentationLink()}
-            target="_blank"
-            external
-          >
-            <FormattedMessage
-              id="xpack.idxMgmt.indexDetails.mappings.docsCardLink"
-              defaultMessage="Learn more about mappings"
-            />
-          </EuiLink>
-        </EuiPanel>
-        {extensionsService.indexMappingsContent && (
-          <>
-            <EuiSpacer />
-            {extensionsService.indexMappingsContent.renderContent({ index, getUrlForApp })}
-          </>
+      {data}
+    </EuiCodeBlock>
+  );
+
+  const treeViewBlock = (
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        {mappingsDefinition === null ? (
```

Comment:
If I look at line 58 above, it looks to me like `mappingsDefinition === null` is a valid state we already check for. In that case the mappings aren't usable for display, but the mappings editor(in `mappings_editor.tsx`) shows a warning for this case that is already available as a component: https://github.com/elastic/kibana/blob/main/x-pack/plugins/index_management/public/application/components/mappings_editor/components/multiple_mappings_warning.tsx

We should just show that warning, I think. We should also change the check to look for `multipleMappingsDeclared` instead of checking for null, because we already check for null up top. `multipleMappingsDeclared` gets set to `true` if `mappingsDefinition` is `null`.

cc @ElenaStoeva

Reply:
@sphilipse Sorry. I missed this comment and merged the PR. Let me address this comment in a separate PR.

Reply:
Created a tech-debt: https://github.com/elastic/enterprise-search-team/issues/7152